### PR TITLE
Add .NET standard support

### PR DESCRIPTION
**Pull request by jeremymeng**
- Upgrade to NUnit v3
- Rewrite http transport using HttpClient
- Add projects targeting .NET STandard
- Update the build scripts and .nuspec files

App.config is not supported on .NET Standard.

For #39 

This PR needs the new tooling: https://www.microsoft.com/net/core#windows. 

I have successful appveyor builds on https://ci.appveyor.com/project/jeremymeng/loggly-csharp.

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -67,8 +67,20 @@ function buildSolution{
...
     Write-Host "Execute Tests"
-    $nunitConsole = "$rootFolder\packages\NUnit.Runners.2.6.3\tools\nunit-console.exe"
-    & $nunitConsole .\Source\Loggly.Tests\bin\$configuration\Loggly.Tests.dll
+
+    $testResultformat = ""
+    $nunitConsole = "$rootFolder\packages\NUnit.ConsoleRunner.3.4.1\tools\nunit3-console.exe"
+
+    if(Test-Path Env:\APPVEYOR){
+        $testResultformat = ";format=AppVeyor"
+        $nunitConsole = "nunit3-console"
```
Comment (jeremymeng) [inline build.ps1]: AppVeyor seems to have a customized runner that report test status in real-time to their service.

File: source/Loggly.Tests/Loggly.Tests.csproj [hunk 2]
```diff
@@ -31,11 +31,40 @@
...
+    <Reference Include="Mono.Cecil.Pdb, Version=0.9.6.0, Culture=neutral, PublicKeyToken=0738eb9f132ed756, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\NUnit3TestAdapter.3.4.1\lib\Mono.Cecil.Pdb.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Mono.Cecil.Rocks, Version=0.9.6.0, Culture=neutral, PublicKeyToken=0738eb9f132ed756, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\NUnit3TestAdapter.3.4.1\lib\Mono.Cecil.Rocks.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
     <Reference Include="Moq">
       <HintPath>..\..\packages\Moq.4.2.1409.1722\lib\net40\Moq.dll</HintPath>
```
Comment (jeremymeng) [inline source/Loggly.Tests/Loggly.Tests.csproj]: If you like, I can also upgrade Moq to the same version as used by the netstandard test project.

File: source/Loggly/Loggly.nuspec [hunk 3]
```diff
@@ -1,18 +1,38 @@
...
+            </group>
+            <group targetFramework=".NETStandard1.5">
+                <dependency id="System.Diagnostics.Process" version="4.1.0" />
+                <dependency id="System.Net.Http" version="4.1.0" />
+                <dependency id="System.Net.NameResolution" version="4.0.0" />
+                <dependency id="System.Net.Requests" version="4.0.11" />
+                <dependency id="System.Net.Security" version="4.0.0" />
+                <dependency id="System.Net.Sockets" version="4.1.0" />
+                <dependency id="System.Diagnostics.Tracing" version="4.1.0" />
+                <dependency id="loggly-csharp-config-netcore" version="$version$" />
```
Comment (jeremymeng) [inline source/Loggly/Loggly.nuspec]: The versions of these two packages should be the same for this to work.  This should be the case when building using `build.ps1`.

**Comment by jeremymeng** [comment]
I have tested the net45 build using the `Loggly.Examples` WinForm app, and did spot checks on the netstandard build using an ASP.NET Core project.

File: source/Loggly.Tests/Loggly.Tests.csproj (hunk 2 above)
Comment (neutmute) [inline source/Loggly.Tests/Loggly.Tests.csproj]: thanks for the contribution! Upgrading moq sounds good

File: source/Loggly/Loggly.nuspec [hunk 4]
```diff
@@ -1,18 +1,38 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
     <metadata>
-        <id>loggly-csharp</id>
+        <id>loggly-csharp-netcore</id>
```
Comment (jeremymeng) [inline source/Loggly/Loggly.nuspec]: This "-netcore" suffix is temporarily for my private build.

File: source/Loggly.Tests/Loggly.Tests.csproj (hunk 2 above)
Comment (jeremymeng) [inline source/Loggly.Tests/Loggly.Tests.csproj]: Done!

**Comment by neutmute** [comment]
This looks good except...it compiles OK in VS2015 but fails from the `build.ps1` script. 
(I installed the two tooling packages - Update 3, .NET Core 1.0.0 - VS 2015 Tooling Preview 2)

Any ideas?

lcs

The text form of the errors in the screenshot is

```

Build FAILED.

"C:\CodeOther\loggly-csharp\loggly.sln" (default target) (1) ->
"C:\CodeOther\loggly-csharp\source\NetStandard\Loggly\Loggly.NetStandard.xproj" (default target) (5) ->
  C:\CodeOther\loggly-csharp\source\NetStandard\Loggly\Loggly.NetStandard.xproj(7,3): error MSB4019: The imported project "C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v12.0\DotNet\Microsoft.DotNet.
Props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.


"C:\CodeOther\loggly-csharp\loggly.sln" (default target) (1) ->
"C:\CodeOther\loggly-csharp\source\NetStandard\Loggly.Config\Loggly.Config.NetStandard.xproj" (default target) (6) ->
  C:\CodeOther\loggly-csharp\source\NetStandard\Loggly.Config\Loggly.Config.NetStandard.xproj(7,3): error MSB4019: The imported project "C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v12.0\DotNet\Mic
rosoft.DotNet.Props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.


"C:\CodeOther\loggly-csharp\loggly.sln" (default target) (1) ->
"C:\CodeOther\loggly-csharp\source\NetStandard\Loggly.Tests\Loggly.Tests.NetStandard.xproj" (default target) (7) ->
  C:\CodeOther\loggly-csharp\source\NetStandard\Loggly.Tests\Loggly.Tests.NetStandard.xproj(7,3): error MSB4019: The imported project "C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v12.0\DotNet\Micro
soft.DotNet.Props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.

    0 Warning(s)
    3 Error(s)
```

It seems unusual it picked v12 and not v14 in those search paths. The msbuild is definitely pointing to v14 though (line 17 build.ps1) yet it says (during nuget download??)

**Comment by jeremymeng** [comment]
Were you building from a VS 2015 command prompt or a VS 2013 command prompt?

Sent from Outlook Mobile<https://aka.ms/blhgte>.

________________________________

**Comment by neutmute** [comment]
doh - good thinking. that's probably it. stale VS params fed into console2.

**Comment by neutmute** [comment]
Looks good! Thanks again! 
I'll remove the `-netcore` suffix in the nuspec before merging to master and then publish as v4.6

**Comment by jeremymeng** [comment]
Thank you!

**Pull request merged**